Feature/m0081 meja 1662 base changelog (#508)
MicroEJ/docs: VEEPortingGuide/architecture7Migration.rst

.. _architecture7_migration:

Architecture 7.x Migration
==========================

This chapter describes the steps to migrate a VEE Port from Architecture ``7.x`` to Architecture ``8.0.0``.

> As a reminder, refer to the :ref:`Architecture 8.0.0 Changelog <changelog-8.0.0>` section for getting the full list of changes and updates.

Reviewer: ```suggestion
As a reminder, refer to the :ref:`Architecture 8.0.0 Changelog <changelog-8.0.0>` section for the complete list of changes and updates.
```

Update Platform Configuration Additions
---------------------------------------

Architecture ``8.0.0`` now directly integrates the :ref:`bsp_connection` mechanism.
The Platform Configuration Additions must be updated accordingly. 

> Contact `our support team <https://www.microej.com/contact/#form_2>`_ to get the detailed instructions. 

Reviewer: ```suggestion
Contact `our support team <https://www.microej.com/contact/#form_2>`_ to get detailed instructions. 
```

Update BSP with new Sections Names
----------------------------------

> The Core Engine sections have been renamed to respect the standard ELF convention. See :ref:`Core Engine Link <core_engine_link>` section.

Reviewer: ```suggestion
The Core Engine sections have been renamed to respect the standard ELF convention. See :ref:`Core Engine Link <core_engine_link>` section for further details.
```

This is usually only used in your linker script file, but section names are sometimes also hardcoded in the C Code.
Here is an example of GNU LD script that highlights the typical changes that must be done:

.. figure:: images/ExampleLD-Architecture7diff8.png
   :alt: Example of LD Script File Migration
   :align: center
   :scale: 100%

   Example of LD Script File Migration

Migrate Built-in Modules
------------------------

The following built-in legacy modules have been removed from the Architecture:

- Device
- ECOM-COMM

In the Platform Editor, these modules now appear in gray with Architecture ``8.x``:

.. figure:: images/architecture7_removed_modules.png
   :scale: 100%

To remove these modules, open the ``.platform`` file using a text editor and remove the following XML elements:

.. code-block:: xml

    <group name="device"/>
    <group name="ecom"/>

Migrate Device Module
~~~~~~~~~~~~~~~~~~~~~

The latest `Device Pack`_ available on the :ref:`central_repository` is backward compatible with the built-in Architecture module.

.. _Device Pack: https://repository.microej.com/modules/com/microej/pack/device/device-pack/

> The following dependency must be added the :ref:`module.ivy <mmm_module_description>` of the VEE Port configuration project:

Reviewer: ```suggestion
The following dependency must be added to the :ref:`module.ivy <mmm_module_description>` of the VEE Port configuration project:
```

      
  .. code:: xml
      
     <dependency org="com.microej.pack.device" name="device-pack" rev="1.1.1" />

Migrate ECOM-COMM Module
~~~~~~~~~~~~~~~~~~~~~~~~

There are two options:
  
- either migrate to the latest ECOM-COMM Pack,

> - or integrate the legacy ECOM-COMM Pack files as-is into your VEE Port ``dropins`` folder,

Reviewer: ```suggestion
- or integrate the legacy ECOM-COMM Pack files as-is into your VEE Port ``dropins`` folder.
```
